src/main/scala/com/campudus/tableaux/router/auth/permission.scala

```diff
@@ -0,0 +1,305 @@
+
+case object Deny extends PermissionType
+
+sealed trait Action {
+  val name: String
+  override def toString: String = name
+}
+
+object Action {
+
+  def apply(action: String): Action = {
+    action match {
+      case Create.name => Create
+      case View.name => View
+      case Edit.name => Edit
+      case Delete.name => Delete
+      case EditDisplayProperty.name => EditDisplayProperty
+      case CreateRow.name => CreateRow
+      case DeleteRow.name => DeleteRow
+      case EditStructureProperty.name => EditStructureProperty
+      case EditCellAnnotation.name => EditCellAnnotation
+      case EditRowAnnotation.name => EditRowAnnotation
+      case ViewCellValue.name => ViewCellValue
+      case EditCellValue.name => EditCellValue
+      case _ => throw new IllegalArgumentException(s"Invalid argument for PermissionType $action")
+    }
+  }
+}
+
+case object Create extends Action { override val name = "create" }
+case object View extends Action { override val name = "view" }
+case object Edit extends Action { override val name = "edit" } // only for media
+case object Delete extends Action { override val name = "delete" }
+case object EditDisplayProperty extends Action { override val name = "editDisplayProperty" }
+case object CreateRow extends Action { override val name = "createRow" }
+case object DeleteRow extends Action { override val name = "deleteRow" }
+case object EditStructureProperty extends Action { override val name = "editStructureProperty" }
+case object EditCellAnnotation extends Action { override val name = "editCellAnnotation" }
+case object EditRowAnnotation extends Action { override val name = "editRowAnnotation" }
+case object ViewCellValue extends Action { override val name = "viewCellValue" }
+case object EditCellValue extends Action { override val name = "editCellValue" }
+
+sealed trait Scope {
+  val name: String
+  override def toString: String = name
+}
+
+object Scope {
+
+  def apply(scope: String): Scope = {
+    scope match {
+      case ScopeTable.name => ScopeTable
+      case ScopeColumn.name => ScopeColumn
+      case ScopeRow.name => ScopeRow
+      case ScopeCell.name => ScopeCell
+      case ScopeMedia.name => ScopeMedia
+      case _ => throw new IllegalArgumentException(s"Invalid argument for PermissionType $scope")
+    }
+  }
+}
+
+case object ScopeTable extends Scope { override val name = "table" }
+case object ScopeColumn extends Scope { override val name = "column" }
+case object ScopeRow extends Scope { override val name = "row" }
+case object ScopeCell extends Scope { override val name = "cell" }
+case object ScopeMedia extends Scope { override val name = "media" }
+
+object ConditionContainer {
+
+  def apply(jsonObjectOrNull: JsonObject): ConditionContainer = {
+
+    val jsonObject: JsonObject = Option(jsonObjectOrNull).getOrElse(Json.emptyObj())
+
+    val ct: ConditionOption = Option(jsonObject.getJsonObject("table")).map(ConditionTable).getOrElse(NoneCondition)
```

Comment:
Please don't use this short variable names. It's not readable for other people. Please name them like conditionTable = Option... and so on.